### PR TITLE
Adapt localstack.dev.kubernetes script to new src layout

### PR DESCRIPTION
**Pull request by dominikschubert**
## Motivation

Recent changes when migrating to a src layout in pro have broken the `localstack.dev.kubernetes` CLI. This PR fixes this by adapting the path to the new nesting structure.

## Changes

- Fixed mounting of pro code into k8s cluster via `localstack.dev.kubernetes` script.
- Fixed mounting of community entrypoints

## Testing
- `python -m localstack.dev.kubernetes --pro` (and check the volume mount paths in the output)

### CODE REVIEW

File: localstack-core/localstack/dev/kubernetes/__main__.py [hunk 1]
```diff
@@ -17,7 +17,9 @@ def generate_k8s_cluster_config(pro: bool = False, mount_moto: bool = False, por
         }
     )
 
-    egg_path = os.path.join(root_path, "localstack_core.egg-info/entry_points.txt")
+    egg_path = os.path.join(
+        root_path, "localstack-core", "localstack_core.egg-info/entry_points.txt"
+    )
```
Comment (alexrashed) [inline localstack-core/localstack/dev/kubernetes/__main__.py:22]: q: Interesting, this seems to be a leftover from #10931. Which means the entrypoint mount wasn't working?

**Review by alexrashed — APPROVED** [review]
Awesome, thanks for adjusting the paths here after the adjustment of the source path initiated by #10800. 🦸🏽

**Comment by alexrashed** [comment]
The test failures in the pipeline are unrelated, I think we can just move on and merge this PR! 🥳

File: localstack-core/localstack/dev/kubernetes/__main__.py (hunk 1 above)
Comment (dominikschubert) [reply localstack-core/localstack/dev/kubernetes/__main__.py:22]: Yep, noticed this coincidentally when checking the generated paths

**Pull request merged**